src/main/java/ru/yandex/clickhouse/ClickHousePreparedStatementFactory.java

```diff
@@ -0,0 +1,60 @@
+package ru.yandex.clickhouse;
+
+
+import org.apache.http.impl.client.CloseableHttpClient;
+import ru.yandex.clickhouse.settings.ClickHouseProperties;
+
+import java.sql.SQLException;
+import java.util.TimeZone;
+
+final class ClickHousePreparedStatementFactory {
+    private static final String INSERT = "insert";
+
+    static ClickHousePreparedStatement getQuery(String query,
+                                                CloseableHttpClient httpClient,
+                                                ClickHouseConnection connection,
+                                                ClickHouseProperties properties,
+                                                TimeZone timeZone) throws SQLException {
+
+        String queryWithoutComments = removeCommentsFrom(query);
+        if (queryWithoutComments.regionMatches(true, 0, INSERT, 0, INSERT.length()))
+            return new ClickHousePreparedInsertStatementImpl(httpClient, connection, properties, queryWithoutComments, timeZone);
+        else
+            return new ClickHousePreparedStatementImpl(httpClient, connection, properties, queryWithoutComments, timeZone);
+
+    }
+
+    static String removeCommentsFrom(String str) {
```

Comment:
Quotation is not handled, and more or less the same logic is there ru.yandex.clickhouse.ClickHousePreparedStatementImpl#parseSql

Also I don't think removing comments is a good idea because it makes it harder to find queries in query_log and trace execution. I think that queries must be modified as little as possible.